### PR TITLE
pkp/pkp-lib#723 Replace citationFormat plugins with CSL integration

### PR DESCRIPTION
**Pull request by NateWr**
Most of the work is in a new plugin repository: https://github.com/pkp/citationStyleLanguage

### CODE REVIEW

File: templates/frontend/objects/article_details.tpl [hunk 1]
```diff
@@ -242,36 +242,48 @@
...
+									{if count($citationDownloads)}
+										<div class="label">
+											{translate key="submission.howToCite.downloadCitation"}
+										</div>
+										<ul class="citation_formats_styles">
+											{foreach from=$citationDownloads item="citationDownload"}
+												<li>
+													<a href="{url page="citationstylelanguage" op="download" path=$citationDownload.id params=$citationArgs}">
+														<span class="fa fa-download"></span>
+														{$citationDownload.title}
```
Comment (asmecher) [inline templates/frontend/objects/article_details.tpl]: Needs `|escape` filter.

File: templates/frontend/objects/article_details.tpl [hunk 2]
```diff
@@ -242,36 +242,48 @@
...
+							<div class="citation_formats">
+								<button class="cmp_button citation_formats_button" aria-controls="cslCitationFormats" aria-expanded="false" data-csl-dropdown="true">
+									{translate key="submission.howToCite.citationFormats"}
+								</button>
+								<div id="cslCitationFormats" class="citation_formats_list" aria-hidden="true">
+									<ul class="citation_formats_styles">
+										{foreach from=$citationStyles item="citationStyle"}
+											<li>
+												<a aria-controls="citationOutput" href="{url page="citationstylelanguage" op="get" path=$citationStyle.id params=$citationArgs}" data-load-citation>
+													{$citationStyle.title}
```
Comment (asmecher) [inline templates/frontend/objects/article_details.tpl]: `|escape`

File: plugins/pubIds/doi/DOIPubIdPlugin.inc.php [hunk 3]
```diff
@@ -18,6 +18,18 @@
...
 class DOIPubIdPlugin extends PubIdPlugin {
 
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	public function register($category, $path) {
+		$success = parent::register($category, $path);
+		if (!Config::getVar('general', 'installed') || defined('RUNNING_UPGRADE')) return $success;
+		if ($success && $this->getEnabled()) {
+			HookRegistry::register('CitationStyleLanguage::citation', array($this, 'getCitationData'));
```
Comment (asmecher) [inline plugins/pubIds/doi/DOIPubIdPlugin.inc.php]: Where is this hook called? (Is there some uncommitted code?)

Comment (asmecher) [reply plugins/pubIds/doi/DOIPubIdPlugin.inc.php]: (Ah, found it: https://github.com/pkp/citationStyleLanguage/blob/master/CitationStyleLanguagePlugin.inc.php#L354)

File: templates/frontend/objects/article_details.tpl (hunk 2 above)
Comment (NateWr) [reply templates/frontend/objects/article_details.tpl]: 8621643cb972b662b49ae4d51b2437a0349c50a1

File: templates/frontend/objects/article_details.tpl (hunk 1 above)
Comment (NateWr) [reply templates/frontend/objects/article_details.tpl]: 8621643cb972b662b49ae4d51b2437a0349c50a1

**Pull request merged**